tauri/src/networks/mod.rs

```diff
@@ -65,24 +64,46 @@ impl Networks {
     }
 
     pub fn get_current_network(&self) -> &Network {
+        if !self.networks.contains_key(&self.current) {
+            return self.networks.values().next().unwrap();
+        }
+
         &self.networks[&self.current]
     }
 
-    pub fn set_networks(&mut self, networks: Vec<Network>) {
-        self.networks = networks.into_iter().map(|n| (n.name.clone(), n)).collect();
-        // TODO: ensure current network still exists
-        self.save().unwrap();
+    pub fn set_networks(&mut self, networks: Vec<Network>) -> Result<()> {
+        let networks = networks.into_iter().map(|n| (n.name.clone(), n)).collect();
+
+        self.do_set_networks(networks)
     }
 
-    pub fn reset_networks(&mut self) {
-        self.networks = Network::default();
-        self.save().unwrap();
+    pub fn reset_networks(&mut self) -> Result<()> {
+        self.do_set_networks(Network::default())
     }
 
     pub fn get_current_provider(&self) -> Provider<Http> {
         self.get_current_network().get_provider()
     }
 
+    fn broadcast_network_changed(&self) -> Result<()> {
```

Comment:
I don't like this name, because it's mixing nomenclature with the `broadcast_chain_changed` which is part of EIP-1193
this is doing 2 different things. that EIP-1193 part, plus notifying our own frontend

how afte `after_network_changed` or `on_network_changed`?

Reply:
👍 sure. I didn't think too much about it because it's a private method